### PR TITLE
feat(postgres): sim support

### PR DESCRIPTION
**Pull request by Chriscbr**
Closes https://github.com/winglang/wing/issues/4914

### CODE REVIEW

File: postgres/lib.test.w [hunk 1]
```diff
@@ -2,11 +2,10 @@ bring expect;
...
-if util.env("WING_TARGET") == "tf-aws" {
-  let db = new l.Database(name: "test", pgVersion: 15);
+let db = new l.Database(name: "test", pgVersion: 15);
 
-  test "run a simple query" {
-    let result = db.query("SELECT 1 as one, 2 as two;");
-    expect.equal(result.at(0), {one: 1, two: 2});
-  }
+test "run a simple query" {
+  util.sleep(1s);
```
Comment (eladb) [inline postgres/lib.test.w:8]: Why is this needed?

File: postgres/lib.w [hunk 2]
```diff
@@ -45,16 +42,52 @@ pub class Database {
   }
 }
 
-pub class DatabaseNeon impl IDatabase {
+class DatabaseSim impl IDatabase {
```
Comment (eladb) [inline postgres/lib.w:45]: Maybe put under `db.sim.w`

File: postgres/lib.w [hunk 3]
```diff
@@ -45,16 +42,52 @@ pub class Database {
...
+    let container = new containers.Workload(
+      name: "postgres",
+      image: image,
+      env: {
+        POSTGRES_PASSWORD: "password"
+      },
+      port: 5432,
+      public: true,
+    );
+    this.url = container.getPublicUrl();
```
Comment (eladb) [inline postgres/lib.w:58]:
```suggestion
    this.url = container.publicUrl;
```

File: postgres/lib.w [hunk 4]
```diff
@@ -45,16 +42,52 @@ pub class Database {
...
+}
+
+struct DbCredentials {
+  host: str;
+  user: str;
+  password: str;
+  database: str;
+}
+
+class DatabaseNeon impl IDatabase {
```
Comment (eladb) [inline postgres/lib.w:81]: `db.neon.w'

**Review by eladb — APPROVED** [review]
Super nice

File: postgres/lib.test.w (hunk 1 above)
Comment (Chriscbr) [reply postgres/lib.test.w:8]: Ah I think I was debugging something, it should be possible to remove it

File: postgres/lib.w (hunk 3 above)
Comment (Chriscbr) [reply postgres/lib.w:58]: I was curious what's the difference between them

Comment (eladb) [reply postgres/lib.w:58]: It's because Interfaces don't support fields and I wanted to enforce the api.

Pretty ugly.

File: postgres/lib.test.w (hunk 1 above)
Comment (Chriscbr) [reply postgres/lib.test.w:8]: Looks like we do need some kind of command-based readiness check.. for now I'm just retrying connecting

File: postgres/lib.w (hunk 2 above)
Comment (eladb) [reply postgres/lib.w:45]: Yes, in other projects we needed to put the interface into a separate file.

I am okay to leave as is

Comment (Chriscbr) [reply postgres/lib.w:45]: Props would have to go in a separate file as well

Comment (eladb) [reply postgres/lib.w:45]: crap
